### ISSUE TITLE
Add configuration-cache flag to control console output

### ISSUE THREAD

**Issue by TWiStErRob** [gradle/gradle#24435]
### Context
A normal, idiomatic Gradle build never outputs anything other than BUILD SUCCESSFUL. Configuration Cache violates this by outputting a lot of state information to the console about caching. The undocumented quiet flag (https://github.com/gradle/gradle/issues/24424) is useful in suppressing some of this, but it's still too noisy because it doesn't have effect on everything it outputs. As devs, we don't care if configuration cache was reused or not, unless we're doing build engineering. For everyday developers, the builds are just faster with configuration cache, and there's no need to link to the report that 0.01% of them will ever open during normal everyday operation.

We've made attempts at understanding and silencing configuration cache: https://github.com/gradle/gradle/issues/19069#issuecomment-1296194155, but it's just not possible to a full extent.

### Current Behavior
on an empty project with configuration cache enabled running `gradlew`:
```
Configuration cache is an incubating feature.
Calculating task graph as no configuration cache is available for tasks: 

...

0 problems were found storing the configuration cache.

See the complete report at file://.../configuration-cache-report.html
```

### Expected Behavior
No logging whatsoever related to configuration cache on most builds, to mirror the behavior when the cache is off. There are no other features that log this heavily. Even test reports, which are actually useful to know where they, are only output location on failure.

So ideally we would have:
```
org.gradle.configuration-cache.verbose=false
```
as the default, and then if someone really needs these log lines, they can turn up the logging.

The currently undocumented `org.gradle.unsafe.configuration-cache.quiet=true` flag can be deprecated against this one asking build engineers to either opt in to verbose, or remove the flag as the quiet mode would be the default.

**Comment by TWiStErRob** [comment]
@abstratt can you please label this.

**Comment by mlopatkin** [comment]
As of Gradle 8.3, we've reduced the output to "Calculating task graph"/"Reusing configuration cache" and "Configuration cache entry stored". We see these to be used in tests inside and outside Gradle, so removing them at this point might be a breaking change.

We'll reconsider this for Gradle 9.0.

**Comment by reinsch82** [comment]
The issue is in the backlog of the relevant team and is prioritized by them.

***

While this improved, we still think we could reduce the verbosity.

**Comment by eskatos** [comment]
> As devs, we don't care if configuration cache was reused or not, unless we're doing build engineering.

Yes!

**Comment by eskatos** [comment]
We agree that the configuration cache output at the beginning and end of build is unnecessary noise. Ideally we get rid of it.

There might be value for this information at INFO level though.

Console output is not considered public API but changing this might impact tools that rely on the current output.
If we want to be careful we could do the change on a major.

@gradle/bt-configuration-cache-project, @gradle/bt-cortex, when would it be a good time to do this? in any release or a major?

**Comment by mlopatkin** [comment]
> We agree that the configuration cache output at the beginning and end of build is unnecessary noise. Ideally we get rid of it.

I disagree that it is unnecessary. There is certain value in being notified about CC state even if you're not working on CC compatibility right now. It at least gives a signals about cache hit rate - allows discovering accidental misses due to excessive inputs. This can be hard to notice if there is no indication, especially if your configuration time is non-trivial but not overwhelming or the execution phase is also slow.

There is a precedent of having task status: `46 actionable tasks: 1 executed, 45 up-to-date`, though I know some folks want to get rid of it too 😄 

We can still agree that we can be more concise. One idea that we've discussed is folding basic CC info into the task status line. Ideally, we should also consider isolated projects there. For example, Gradle may say:
1. No cached state: `Configured 100500 projects; 46 actionable tasks: 1 executed, 45 up-to-date`
2. Cache miss: `Reconfigured 100500 projects, 0 reused; ...`
3. Partial hit with IP: `Reconfigured 5 projects, 1337 reused; ...`
4. Full hit: `100500 projects reused; ...`

Users who are interested in looking further may then look into the CC report or a build scan.

> There might be value for this information at INFO level though.
>
>Console output is not considered public API but changing this might impact tools that rely on the current output. If we want to be careful we could do the change on a major.

Our current recommendation for testing CC is to [check the output](https://docs.gradle.org/current/userguide/configuration_cache_debugging.html#config_cache:testing:~:text=output.contains%28%22Reusing%20configuration%20cache.%22%29), so we'll be breaking all tests in the wild. With the verbosity `INFO` brings, I'd rather not see us forcing people to enable it in tests.

I strongly believe we need better testing story before we can take an action here.

> [@gradle/bt-configuration-cache-project](https://github.com/orgs/gradle/teams/bt-configuration-cache-project), [@gradle/bt-cortex](https://github.com/orgs/gradle/teams/bt-cortex), when would it be a good time to do this? in any release or a major?

Due to the effect on testing, I'd vote for a major release.

**Comment by eskatos** [comment]
IIUC we need a better testing story before we can make this change in a major.
Is there an issue about this we can block this one on?

**Comment by mlopatkin** [comment]
> IIUC we need a better testing story before we can make this change in a major. Is there an issue about this we can block this one on?

Actually, there is:
* https://github.com/gradle/gradle/issues/35595
